src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/FallThroughCheck.java

```diff
@@ -486,25 +495,25 @@ private boolean checkSynchronized(final DetailAST synchronizedAst, boolean useBr
      * </pre>
      *
      * @param currentCase AST of the case that falls through to the next case.
-     * @param nextCase AST of the next case.
      * @return True if a relief comment was found
      */
-    private boolean hasFallThroughComment(DetailAST currentCase, DetailAST nextCase) {
-        boolean allThroughComment = false;
-        final int endLineNo = nextCase.getLineNo();
-
-        if (matchesComment(reliefPattern, endLineNo)) {
-            allThroughComment = true;
+    private boolean hasFallThroughComment(DetailAST currentCase) {
+        final DetailAST nextCase = currentCase.getNextSibling();
+        DetailAST commentAst;
+        if (nextCase.getType() == TokenTypes.CASE_GROUP) {
+            commentAst = nextCase.getFirstChild();
```

Comment:
this is not always true.
comment location in case is unpredictable.
please use util method to find comment node under Case node.

Reply:
please check again